testsuite/tests/typing-modules/records_errors_test.ml

```diff
@@ -134,5 +144,143 @@ Error: Signature mismatch:
          type t = { f0 : unit; }
        is not included in
          type t = { f0 : unit; f1 : unit; }
-       The field f1 is only present in the second declaration.
+       A field, f1, is missing in the first declaration.
 |}];;
+
+
+(** Random additions and deletions of fields *)
+
+module Addition : sig
+  type t = {a : unit; b : unit; c : unit; d : unit}
+end = struct
+  type t = {a : unit; b : unit; beta : unit; c : unit; d: unit}
+end
+[%%expect {|
+Lines 5-7, characters 6-3:
+5 | ......struct
+6 |   type t = {a : unit; b : unit; beta : unit; c : unit; d: unit}
+7 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig
+           type t = { a : unit; b : unit; beta : unit; c : unit; d : unit; }
+         end
+       is not included in
+         sig type t = { a : unit; b : unit; c : unit; d : unit; } end
+       Type declarations do not match:
+         type t = { a : unit; b : unit; beta : unit; c : unit; d : unit; }
+       is not included in
+         type t = { a : unit; b : unit; c : unit; d : unit; }
```

Comment:
The "modules do not match" and "type declarations do not match" are painfully redundant. I understand that this is partly an orthogonal issue to your change, but are there plans to solve that part as well?

Reply:
This is more a general question on how to print the error tree. Currently, we are emitting a full message at every step from the root of the tree to the leaf on focus. This leads to repetitions since the context at step `k` does not change much from the context at step `k+1`.  I am planing to propose another strategy (print a full context at most once), but that will be a separate PR.

Reply:
Sure, this should be handled separately.